cumulus/pallets/collator-selection/src/lib.rs

```diff
@@ -320,8 +342,8 @@ pub mod pallet {
 			// don't wipe out the collator set
 			if new.is_empty() {
 				ensure!(
-					Candidates::<T>::decode_len().unwrap_or_default() >=
-						T::MinEligibleCollators::get() as usize,
+					CandidateList::<T>::decode_len().unwrap_or_default() >=
+						T::MinEligibleCollators::get().try_into().unwrap(),
```

Comment:
These `.try_into().unwrap()` statements would be good candidates for the `integrity_test` hook. Then we can just ensure that it will always work. And as long as `T::MinElegibleCollators` is u32 (usize) then it obviously will.

Reply:
I added a comment as well as an integrity test. PTAL